src/mds/DamageTable.cc

```diff
@@ -200,6 +201,29 @@ bool DamageTable::notify_remote_damaged(inodeno_t ino, std::string_view path)
   return false;
 }
 
+void DamageTable::remove_dentry_damage_entry(CDir *dir)
+{
+  const std::map<DentryIdent, DamageEntryRef> &frag_dentries =
+    dentries.at(DirFragIdent(dir->inode->ino(), dir->frag));
+    for (const auto &i : by_id){
+      for (const auto &dn : frag_dentries){
+        if (i.second == dn.second){
+          DamageTable::erase(i.first);
```

Comment:
I was concerned for the matching algorithm time complexity being O(m x n) and was wondering if it can be optimized.

But maybe there won't be too many damaged entries in the table for this issue to be of real concern.

Reply:
Yup, I too was concerned about the complexity but couldn't think of that's why choose this way unlike others damage types.Will re-think.